### PR TITLE
test supervisor/subjects

### PR DESCRIPTION
**Pull request by PainterHalver**
## Related Tickets
- [#66203](https://edu-redmine.sun-asterisk.vn/issues/66203)

## WHAT (optional)
- Test Supervisor::SubjectsController.

## HOW
- Thêm factory cho model question, test.
- Refactor shared example nhận method get, post...
- Fix bug redirect trong controller.

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by PainterHalver** [comment]
ready

File: app/controllers/supervisor/subjects_controller.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def update
   def destroy
     if has_ongoing_test? @subject
       flash[:danger] = t "has_ongoing_test"
-      redirect_to supervisor_subjects_path
+      return redirect_to supervisor_subjects_path
```
Comment (vanvtt-0952) [inline app/controllers/supervisor/subjects_controller.rb]: nếu ri, sao em ko taọ 1 method cho code if này nhỉ, dùng action filter để call là đẹp thay vì return

**Comment by PainterHalver** [comment]
ready

**Comment by PainterHalver** [comment]
ready
rebase conflict

**Pull request merged**